sdk/core/Azure.Core/tests/AzureSasCredentialSynchronousPolicyTests.cs

```diff
@@ -108,8 +108,8 @@ public async Task VerifyRetryAfterSasCredentialUpdateOperation()
                 if (callCount++ == 0)
                 {
                     Assert.AreEqual("?foo=bar&sig=first_signature_value", req.Uri.Query);
-                    azureSasCredential.Update(SECOND_SIGNATURE_VALUE);
                     Thread.Sleep(100);
+                    azureSasCredential.Update(SECOND_SIGNATURE_VALUE);
```

Comment:
Would you be willing to describe why this fixes the issue?

Reply:
Yes - in certain timing conditions, `message.ProcessingContext.StartTime.Ticks` is equal to `AzureSasCredential.SignatureUpdated`. The placement of this delay ensures that there is some space between the two time values. That is what I intended to do initially, but I put the delay in the wrong place.

Reply:
Thanks!